### PR TITLE
The return of the Jeudi branch DBE

### PR DESCRIPTION
**Pull request by dbeguin**

### CODE REVIEW

File: addons/knowledge/static/src/js/wysiwyg.js [hunk 1]
```diff
@@ -203,15 +203,17 @@ Wysiwyg.include({
...
-            const articleLinkFragment = new DocumentFragment();
-            const articleLinkBlock = $(QWeb.render('knowledge.wysiwyg_article_link', {
-                display_name: data.display_name,
-                href: '/knowledge/article/' + data.id,
-                article_id: data.id,
-            }))[0];
-            articleLinkFragment.append(articleLinkBlock);
-            const [anchor] = this.odooEditor.execCommand('insertFragment', articleLinkFragment);
-            this._notifyNewBehaviors(anchor);
+            if (data) {
```
Comment (tde-banana-odoo) [inline addons/knowledge/static/src/js/wysiwyg.js:206]: Data does not mean anything... where does it come from ? What happens when we have no data, why wouldn't we have some data ? Try to give meaningful variable name (and maybe a comment saying "result coming from user input / call to rpc 'blah' : ...", just to ease the global understanding of the code :)

cc @abd-msyukyu-odoo

File: addons/knowledge/static/src/webclient/commands/knowledge_providers.js [hunk 2]
```diff
@@ -11,46 +11,46 @@ const { Component, xml } = owl;
...
-        <icon t-if="props.isFavorite" class="fa fa-star o_favorite pr-2 position-static"/>
-        <span t-if="props.subjectName" t-out="'— '" class="text-muted small pr-2" />
-        <span t-if="props.subjectName" class="o_command_name text-muted small">
-            <t t-foreach="props.splitSubjectName" t-as="name" t-key="name_index">
-                <b t-if="name_index % 2" t-out="name"/>
-                <t t-else="" t-out="name"/>
-            </t>
-        </span>
-    </div>
+<div class="o_command_left d-flex align-items-center">
```
Comment (tde-banana-odoo) [inline addons/knowledge/static/src/webclient/commands/knowledge_providers.js:14]: Why do we have (badly indented) xml in a js file ?

cc @abd-msyukyu-odoo

File: addons/knowledge/static/src/webclient/commands/knowledge_providers.js [hunk 3]
```diff
@@ -11,46 +11,46 @@ const { Component, xml } = owl;
...
 
 // "Not found, create one" command
 class Knowledge404Command extends Component {}
 Knowledge404Command.template = xml`
-    <div class="o_command_hotkey">
-        <span>
-            No Article found. <span class="text-primary">Create "<u t-out="props.articleName"/>"</span>
-        </span>
-    </div>
+<div class="o_command_hotkey">
```
Comment (tde-banana-odoo) [inline addons/knowledge/static/src/webclient/commands/knowledge_providers.js:32]: Same here

File: addons/knowledge/static/src/webclient/commands/knowledge_providers.js [hunk 4]
```diff
@@ -11,46 +11,46 @@ const { Component, xml } = owl;
...
-            <t t-out="props.name" />
-        </span>
-        <span>
-            <t t-foreach="getKeysToPress(props)" t-as="key" t-key="key_index">
-                <kbd t-out="key" />
-                <span t-if="!key_last"> + </span>
-            </t>
-        </span>
-    </div>
+<div class="o_command_hotkey">
```
Comment (tde-banana-odoo) [inline addons/knowledge/static/src/webclient/commands/knowledge_providers.js:42]: Same here

File: addons/knowledge/views/knowledge_article_views.xml [hunk 5]
```diff
@@ -141,13 +143,13 @@
                                                 <!-- Article body -->
                                                 <div class="o_knowledge_body d-flex flex-column">
                                                     <!-- Article Cover -->
-                                                    <field name="cover" widget="image" class="o_knowledge_cover_image" attrs="{'invisible': [('cover', '=', False)]}"/>
+                                                    <field name="cover" widget="image" class="o_knowledge_cover_image" attrs="{'invisible': [('cover', '=', False)], 'readonly': [('is_locked', '=', True)]}"/>
```
Comment (tde-banana-odoo) [inline addons/knowledge/views/knowledge_article_views.xml:146]: Linting: multi line (makes code explode in screen and modifying an attribute make the whole line as changed, hard to spot changes)

File: addons/knowledge/views/knowledge_article_views.xml [hunk 6]
```diff
@@ -141,13 +143,13 @@
...
 
                                                     <!-- Full Width handling -->
                                                     <span class="o_knowledge_article_view_form_dynamic_width d-none" attrs="{'invisible': [('full_width', '=', False)]}"/>
 
                                                     <!-- Icon and Cover buttons -->
-                                                    <div class="text-primary bg-white o_knowledge_icon_cover_buttons mt-2" attrs="{'invisible': ['|', ('is_locked', '=', True), ('user_can_write', '=', False)]}">
+                                                    <div class="o_knowledge_icon_cover_buttons text-primary bg-white d-print-none mt-2" attrs="{'invisible': ['|', ('is_locked', '=', True), ('user_can_write', '=', False)]}">
```
Comment (tde-banana-odoo) [inline addons/knowledge/views/knowledge_article_views.xml:152]: Linting: multi line (makes code explode in screen and modifying an attribute make the whole line as changed, hard to spot changes)

File: addons/knowledge/views/knowledge_article_views.xml [hunk 7]
```diff
@@ -208,7 +210,6 @@
             <tree default_order="is_user_favorite desc, favorite_count desc">
                 <field name="display_name"/>
                 <field name="is_user_favorite" widget="boolean_favorite" nolabel="1" force_save="1" options="{'allow_order': '1'}"/>
-                <field name="favorite_count"/>
```
Comment (tde-banana-odoo) [inline addons/knowledge/views/knowledge_article_views.xml]: Can we order by a field that is not in the view (default order hereabove)

File: addons/knowledge/tests/test_knowledge_article_business.py [hunk 8]
```diff
@@ -343,12 +343,17 @@ def test_article_move_to(self):
         self.assertEqual(workspace_children[1].parent_id, workspace_children[0])
         self.assertEqual(workspace_children[0].parent_id, article_workspace)
 
+        # Test that desynced articles are resynced when moved to root
```
Comment (tde-banana-odoo) [inline addons/knowledge/tests/test_knowledge_article_business.py:445]: Would make a specific unit test. Current one test mainly the parameters (parent_id, is_private). I would make a ``test_article_move_to_desync`` test checking this small change specifically. Purpose is to have small but easy to reproduce and understand tests :) .

**Review by tde-banana-odoo — COMMENTED** [review]
Zbouik !

File: addons/knowledge/views/knowledge_article_views.xml (hunk 7 above)
Comment (fbarbe00) [reply addons/knowledge/views/knowledge_article_views.xml]: Is this a question? The answer is no. Should I bring the field back?

**Pull request closed without merging**